gate-core/src/main/groovy/com/netflix/spinnaker/gate/config/ServiceConfiguration.groovy

```diff
@@ -48,4 +53,28 @@ class ServiceConfiguration {
   Service getService(String name) {
     (services + integrations)[name]
   }
+
+  Endpoint getServiceEndpoint(String serviceName, String dynamicName = null) {
+    Service service = getService(serviceName)
+
+    if (service == null) {
+      throw new IllegalArgumentException("Unknown service ${serviceName}")
+    }
+
+    Endpoint endpoint
+    if (dynamicName == null) {
+      endpoint = discoveryHosts && service.vipAddress ?
+        newFixedEndpoint("niws://${service.vipAddress}")
+        : newFixedEndpoint(service.baseUrl)
```

Comment:
Blech. Not a fan of seeing anything related to NIWS in Spinnaker OSS code. Is there not a way to introduce an interface here so we can keep Netflix-only things in our extension repos?

Reply:
Oh, I see this is just shifting some already leaky code into a different place. I guess could you just put a `TODO` comment here that we need to comeback and clean this up at some point?

Reply:
I didn't add this. I just extracted it into a method...